music/src/core/visualizer.ts

```diff
@@ -456,3 +458,237 @@ export class PianoRollSVGVisualizer extends BaseVisualizer {
+ * (PAGE is default value), note by note (NOTE) or in packed chunks by doing 
+ * scroll just on bar starting (BAR).
+ */
+export enum ScrollType {
+  PAGE = 0,
+  NOTE = 1,
+  BAR = 2
+}
+
+/**
+ * An interface for providing extra configurable properties to a Visualizer
+ * extending the basic configurable properties of `VisualizerConfig`.
+ * @param defaultKey The musical key the score must use to adapt the score to 
+ * the right accidentals. It can be overwritten with 
+ * `NoteSequence.keySignatures` value at time or step 0. If not assigned it 
+ * will be asumed C key.
+ * @param instruments The subset of the `NoteSequence` instrument track numbers 
+ * which should be merged and displayed. If not assigned or equal to [] it will 
+ * be used all instruments altogether.
+ * @param scrollType Sets scrolling to follow scoreplaying in different ways 
+ * according to `ScrollType` enum values.
+ */
+export interface StaffSVGVisualizerConfig extends VisualizerConfig {
+  defaultKey?: number;
+  instruments?: number[];
+  scrollType?: ScrollType;
+}
+
+/**
+ * Displays a `NoteSecuence` as a staff on a given SVG. Staff is scaled to fit 
+ * vertically `config.noteHeight` and note horizontal position can behave in 
+ * two different ways: If `config.pixelsPerTimeStep` is greater than zero, 
+ * horizontal position will be proportional to its starting time, allowing to 
+ * pile several instances for different voices (parts). Otherwise, resulting 
+ * staff will display notes in a compact form, using minimum horizontal space 
+ * between musical symbols as regular paper staff does. 
+ * 
+ * Clef, key and time signature will be displayed at the leftmost side and the 
+ * rest of the staff will scroll under this initial signature area accordingly.
+ * In case of proportional note positioning, given it starts at pixel zero, the
+ * signature area will blink meanwhile it collides with initial active notes.
+ * Key and time signature changes will be shown accordingly through score.
+ * 
+ * New configuration features have been introduced through 
+ * `StaffSVGVisualizerConfig` over basic `VisualizerConfig`.
+ * 
+ * When connected to a player, the visualizer can also highlight 
+ * the notes being currently played.
+ * 
+ * You can find more info at:
+ * 
+ * https://github.com/rogerpasky/staffrender-magentaviewer
+ */
+export class StaffSVGVisualizer extends BaseVisualizer {
+  private render: sr.StaffSVGRender; // The actual render
+  private instruments: number[]; // Instruments filter to be rendered
+  private drawnNotes: number; // Number of drawn notes. Will redraw if changed
+
+  /**
+   * `StaffSVGVisualizer` constructor.
+   *
+   * @param sequence The `NoteSequence` to be visualized.
+   * @param div The element where the visualization should be displayed.
+   * @param config Visualization configuration options.
+   */
+  constructor(
+    sequence: INoteSequence, 
+    div: HTMLDivElement, 
+    config: StaffSVGVisualizerConfig = {}
+  ) {
```

Comment:
The formatting in this code is super weird. Are you using the clang plugin in vscode? (it only works if you just open the root `music` folder)

Reply:
My fault. I've manually changed formatting to adapt to the rest of code:

* Braces: Stroustrup style with 2 spaces indentation
* Long lines: 4 spaces indentation

I tried to get the clang plugin in vscode, but there was several ones (xaver.clang-format, mitaki28.vscode-clang, kube.clangcomplete and others) and I didn't want to mess things on the rest of the file. Please, let me know which one and I'll stick to that for next contributions.